cachi2/core/package_managers/rpm/redhat.py

```diff
@@ -0,0 +1,168 @@
+
+
+class RedhatRpmsLock(UserDict):
+    """
+    The class implements basic operations with specific lockfile format.
+
+    Input data comes from raw yaml stored as a dictionary.
+    """
+
+    # parsed structure of the lockfile
+    _lockfile: Root
+    # metadata of file records found in the lockfile (for verification and SBOM generation)
+    _files_metadata: dict[Path, Any] = {}
+    # 'repoid' key is not mandatory in the lockfile format. When not present,
+    # fallback is set as (partly) random string containing _uuid
+    _uuid = uuid.uuid4().hex[:6]
+
+    @property
+    def internal_repoid(self) -> str:
+        """Internal_repoid getter."""
+        return f"cachi2-{self._uuid}"
+
+    @property
+    def internal_source_repoid(self) -> str:
+        """Internal_source_repoid getter."""
+        return f"cachi2-{self._uuid}-source"
+
+    def is_internal_repoid(self, repoid: str) -> bool:
+        """Repoid directory matches the internal repoid."""
+        return repoid in (self.internal_repoid, self.internal_source_repoid)
+
+    @property
+    def metadata(self) -> dict[Path, Any]:
+        """Metadata getter."""
+        return self._files_metadata
+
+    def match_format(self) -> bool:
+        """Parse lockfile header with pydantic and evaluate whether header \
+        matches the format specification."""
+        try:
+            header = Header(**self.data)
+        except ValidationError:
+            return False
+        return header.lockfileVendor == "redhat" and header.lockfileVersion == 1
+
+    def process_format(self) -> bool:
+        """Parse the whole lockfile with pydantic and return whether \
+        the parsing was successful and format of the lockfile is correct."""
+        try:
+            self._lockfile = Root(**self.data)
+        except ValidationError as e:
+            log.error(str(e))  # log defail of the validation error
+            return False
+        return bool(self._lockfile)
+
+    def download(self, output_dir: Path) -> None:
+        """
+        Download packages mentioned in the lockfile.
+
+        Go through the parsed lockfile structure and find all RPM and SRPM files.
+        Create a metadata structure indexed by destination path used
+        for later verification (size, checksum) after download.
+        Prepare a list of files to be downloaded, and then download files.
+        """
+        for arch in self._lockfile.arches:
+            log.debug(f"Downloading files for '{arch.arch}' architecture.")
+            # files per URL for downloading packages & sources
+            files: dict[str, Union[str, PathLike[str]]] = {}
+            for pkg in arch.packages:
+                repoid = self.internal_repoid if pkg.repoid is None else pkg.repoid
+                dest = output_dir.joinpath(arch.arch, repoid, Path(pkg.url).name)
+                files[pkg.url] = str(dest)
+                self._files_metadata[dest] = {
+                    "package": True,  # RPM
```

Comment:
with this boolean gone as discussed, there's now nothing unique about this `_files_metadata` class attribute that wouldn't already be contained within the dictionary already, we just need to use it.